signer/src/storage/postgres.rs

```diff
@@ -92,8 +97,136 @@ impl TryFrom<&NakamotoBlock> for model::StacksBlock {
 
 impl PgStore {
     /// Connect to the Postgres database at `url`.
-    pub async fn connect(url: &str) -> Result<Self, sqlx::Error> {
-        Ok(Self(sqlx::PgPool::connect(url).await?))
+    pub async fn connect(url: &str) -> Result<Self, Error> {
+        Ok(Self(
+            sqlx::PgPool::connect(url)
+                .await
+                .map_err(Error::SqlxConnect)?,
+        ))
+    }
+
+    /// Apply the migrations to the database.
+    // Note 1: This could be generalized and moved up to the `storage` module, but
+    // left that for a future exercise if we need to support other databases.
+    // Note 2: The `sqlx` "migration" feature results in dependency conflicts
+    // with sqlite from the clarity crate.
+    // Note 3: The migration code paths have no explicit integration tests, but are
+    // implicitly tested by all integration tests using `new_test_database()`.
+    pub async fn apply_migrations(&self) -> Result<(), Error> {
+        tracing::info!("Preparing to run database migrations");
+
+        sqlx::raw_sql(
+            r#"
+                CREATE TABLE IF NOT EXISTS public.__sbtc_migrations (
+                    key TEXT PRIMARY KEY
+                );
```

Comment:
I was thinking that we'd probably want to add a timestamp and a bunch of other stuff here too (basically the columns added by tools like `flyway` and `sqlx-cli`). But most of those are probably overkill. 🤔

Reply:
Yeah I had some but removed them for simplicity's sake at this stage -- I didn't want to add things "just because", figured we could wait until the use-case presented itself.